cli/eth.go

```diff
@@ -0,0 +1,178 @@
+package cli
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-address"
+
+	"github.com/filecoin-project/lotus/api/v0api"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/chain/types/ethtypes"
+)
+
+var EthCmd = &cli.Command{
+	Name:  "eth",
+	Usage: "Query eth contract state",
+	Subcommands: []*cli.Command{
+		EthGetAddressCmd,
+		EthCallSimulateCmd,
+	},
+}
+
+func ethAddrFromFilecoinAddress(ctx context.Context, addr address.Address, fnapi v0api.FullNode) (ethtypes.EthAddress, address.Address, error) {
+	var faddr address.Address
+	var err error
+
+	switch addr.Protocol() {
+	case address.BLS, address.SECP256K1:
+		faddr, err = fnapi.StateLookupID(ctx, addr, types.EmptyTSK)
+		if err != nil {
+			return ethtypes.EthAddress{}, addr, err
+		}
+	case address.Actor, address.ID:
+		f0addr, err := fnapi.StateLookupID(ctx, addr, types.EmptyTSK)
+		if err != nil {
+			return ethtypes.EthAddress{}, addr, err
+		}
+		faddr, err = fnapi.StateAccountKey(ctx, f0addr, types.EmptyTSK)
+		if err != nil {
+			return ethtypes.EthAddress{}, addr, err
+		}
+		if faddr.Protocol() == address.BLS || addr.Protocol() == address.SECP256K1 {
+			faddr = f0addr
+		}
+	case address.Delegated:
+		faddr = addr
+	default:
+		return ethtypes.EthAddress{}, addr, xerrors.Errorf("Filecoin address doesn't match known protocols")
+	}
+
+	ethAddr, err := ethtypes.EthAddressFromFilecoinAddress(faddr)
+	if err != nil {
+		return ethtypes.EthAddress{}, addr, err
+	}
+
+	return ethAddr, faddr, nil
+}
+
+var EthGetAddressCmd = &cli.Command{
+	Name:      "stat",
```

Comment:
Odd name for the command?

Reply:
This was specified by steb. I don't have a strong opinion. Do you have a preferred name?